### ISSUE TITLE
localkube crashing after trying kube-prometheus

### ISSUE THREAD

**Issue by abh** [kubernetes/minikube#1535]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

BUG REPORT

**Minikube version** (use `minikube version`):

0.19

**Environment**:
- **OS** (e.g. from /etc/os-release):

2016.08

- **VM Driver**:

VirtualBox

- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`):

0.18.0

**What happened**:

I installed the prometheus-operator from https://github.com/coreos/prometheus-operator/tree/master/contrib/kube-prometheus (with the instructions there yesterday). Today after rebooting minikube localkube isn't coming up (see crash below).

**What you expected to happen**:

localkube not crashing. :-)

**How to reproduce it** (as minimally and precisely as possible):

Install prometheus-operator, remove (some of it?) again.

**Anything else do we need to know**:

I installed the version 0.0.1 operator from the coreos website first and cleaned it out (I think).

The API is up for a very brief moment, I deleted the "monitoring" namespace in one of those moments but it didn't help.

```
May 29 01:41:14 minikube localkube[13732]: apiserver is ready!
May 29 01:41:14 minikube localkube[13732]: Starting controller-manager...
May 29 01:41:14 minikube localkube[13732]: Waiting for controller-manager to be healthy...
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.160515   13732 leaderelection.go:179] attempting to acquire leader lease...
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.163250   13732 leaderelection.go:189] successfully acquired lease kube-system/kube-controller-manager
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.164693   13732 event.go:217] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"kube-system", Name:"kube-controller-manager", UID:"2cacf3f1-432d-11e7-ac41-0800276c481c", APIVersion:"v1", ResourceVersion:"419506", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' minikube became leader
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.171420   13732 controllermanager.go:437] Started "resourcequota"
May 29 01:41:14 minikube localkube[13732]: W0529 01:41:14.171687   13732 controllermanager.go:421] "tokencleaner" is disabled
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.171608   13732 resource_quota_controller.go:240] Starting resource quota controller
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.172258   13732 controllermanager.go:437] Started "podgc"
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.172691   13732 controllermanager.go:437] Started "job"
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.173100   13732 controllermanager.go:437] Started "deployment"
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.173250   13732 deployment_controller.go:151] Starting deployment controller
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.173692   13732 controllermanager.go:437] Started "horizontalpodautoscaling"
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.173722   13732 horizontal.go:139] Starting HPA Controller
May 29 01:41:14 minikube localkube[13732]: E0529 01:41:14.173952   13732 certificates.go:38] Failed to start certificate controller: open /etc/kubernetes/ca/ca.pem: no such file or directory
May 29 01:41:14 minikube localkube[13732]: W0529 01:41:14.174148   13732 controllermanager.go:434] Skipping "certificatesigningrequests"
May 29 01:41:14 minikube localkube[13732]: W0529 01:41:14.174221   13732 controllermanager.go:421] "bootstrapsigner" is disabled
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.174541   13732 controllermanager.go:437] Started "endpoint"
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.174833   13732 controllermanager.go:437] Started "replicationcontroller"
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.174930   13732 replication_controller.go:150] Starting RC Manager
May 29 01:41:14 minikube localkube[13732]: E0529 01:41:14.175191   13732 util.go:45] Metric for serviceaccount_controller already registered
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.175322   13732 controllermanager.go:437] Started "serviceaccount"
May 29 01:41:14 minikube localkube[13732]: W0529 01:41:14.175467   13732 registered.go:288] Could not find groupMeta for monitoring.coreos.com
May 29 01:41:14 minikube localkube[13732]: I0529 01:41:14.175399   13732 serviceaccounts_controller.go:122] Starting ServiceAccount controller
May 29 01:41:14 minikube localkube[13732]: panic: runtime error: invalid memory address or nil pointer dereference
May 29 01:41:14 minikube localkube[13732]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x17ad772]
May 29 01:41:14 minikube localkube[13732]: goroutine 1196 [running]:
May 29 01:41:14 minikube localkube[13732]: panic(0x37d2ca0, 0xc420016030)
May 29 01:41:14 minikube localkube[13732]:         /usr/local/go/src/runtime/panic.go:500 +0x1a1
May 29 01:41:14 minikube localkube[13732]: k8s.io/minikube/vendor/k8s.io/apimachinery/pkg/apimachinery/registered.(*APIRegistrationManager).RESTMapper(0xc42018e550, 0x0, 0x0, 0x0, 0x43eb3b, 0xc4247a9208)
May 29 01:41:14 minikube localkube[13732]:         /usr/local/google/home/mrick/go/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/k8s.io/apimachinery/pkg/apimachinery/registered/registered.go:290 +0x2a2
May 29 01:41:14 minikube localkube[13732]: k8s.io/minikube/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app.startGarbageCollectorController(0x6918f20, 0xc423de4fc0, 0x6923f40, 0xc42274fc20, 0x0, 0x0, 0x0, 0x0, 0xc4222a8a20, 0x1, ...)
May 29 01:41:14 minikube localkube[13732]:         /usr/local/google/home/mrick/go/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/core.go:166 +0x6d
May 29 01:41:14 minikube localkube[13732]: k8s.io/minikube/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app.StartControllers(0xc420175ec0, 0xc4223922c0, 0x6918f20, 0xc423de4fc0, 0x6918f20, 0xc423de4fc0, 0xc42244d4a0, 0x0, 0x0)
May 29 01:41:14 minikube localkube[13732]:         /usr/local/google/home/mrick/go/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go:428 +0x5b3
May 29 01:41:14 minikube localkube[13732]: k8s.io/minikube/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app.Run.func2(0xc42244d4a0)
May 29 01:41:14 minikube localkube[13732]:         /usr/local/google/home/mrick/go/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go:180 +0xc6
May 29 01:41:14 minikube localkube[13732]: created by k8s.io/minikube/vendor/k8s.io/kubernetes/pkg/client/leaderelection.(*LeaderElector).Run
May 29 01:41:14 minikube localkube[13732]:         /usr/local/google/home/mrick/go/src/k8s.io/minikube/_gopath/src/k8s.io/minikube/vendor/k8s.io/kubernetes/pkg/client/leaderelection/leaderelection.go:150 +0x97
```

**Comment by abh** [comment]
FWIW `kubectl delete  thirdpartyresources  --all` while the API was up got things going again.

**Comment by r2d4** [comment]
This will be fixed by https://github.com/kubernetes/minikube/pull/1497

**Comment by r2d4** [comment]
This is fixed in 0.19.1
